CMakeLists.txt

```diff
@@ -34,22 +34,30 @@ endif()
 
 option(SOLC_LINK_STATIC "Link solc executable statically on supported platforms" OFF)
 option(SOLC_STATIC_STDLIBS "Link solc against static versions of libgcc and libstdc++ on supported platforms" OFF)
-option(STRICT_Z3_VERSION "Use the latest version of Z3" ON)
+option(STRICT_Z3_VERSION "Require the exact version of Z3 solver expected by our test suite." ON)
 option(PEDANTIC "Enable extra warnings and pedantic build flags. Treat all warnings as errors." ON)
 option(PROFILE_OPTIMIZER_STEPS "Output performance metrics for the optimiser steps." OFF)
-option(USE_SYSTEM_LIBRARIES "Use system libraries" OFF)
-option(ONLY_BUILD_SOLIDITY_LIBRARIES "Only build solidity libraries" OFF)
-option(STRICT_NLOHMANN_JSON_VERSION "Strictly check installed nlohmann json version" ON)
```

Comment:
I wanted to rename this to avoid confusion with `STRICT_Z3_VERSION`, which works a bit differently, but then I thought that we might be better off just removing the version check altogether.

It seems to have been added way back in #3913. At that time we were still using jsoncpp, testing with Travis and downloading the library via CMake. Now all of that has changed and I think that the risk of getting the wrong version is much lower, especially with it now coming from a submodule.

nlohmann JSON is supposed to be more resilient and strict than jsoncpp so hopefully any version that builds and passes our test suite is fine. We should not have to enforce an exact version. We don't do that for fmtlib or ranges either.